navigator/backend/src/main/scala/com/digitalasset/navigator/json/ApiCodecVerbose.scala

```diff
@@ -91,31 +100,29 @@ object ApiCodecVerbose {
   def apiVariantToJsValue(value: Model.ApiVariant): JsValue =
     JsObject(
       propType -> JsString(tagVariant),
-      propId -> value.variantId.map(_.toJson).getOrElse(JsNull),
-      propConstructor -> JsString(value.constructor),
+      propId -> value.tycon.map(_.toJson).getOrElse(JsNull),
+      propConstructor -> JsString(value.variant),
       propValue -> apiValueToJsValue(value.value)
     )
 
   def apiEnumToJsValue(value: Model.ApiEnum): JsValue =
     JsObject(
       propType -> JsString(tagEnum),
-      propId -> value.enumId.map(_.toJson).getOrElse(JsNull),
-      propConstructor -> JsString(value.constructor),
+      propId -> value.tycon.map(_.toJson).getOrElse(JsNull),
+      propConstructor -> JsString(value.value),
     )
 
   def apiRecordToJsValue(value: Model.ApiRecord): JsValue =
     JsObject(
       propType -> JsString(tagRecord),
-      propId -> value.recordId.map(_.toJson).getOrElse(JsNull),
-      propFields -> JsArray(
-        value.fields
-          .map(
-            f =>
-              JsObject(
-                propLabel -> JsString(f.label),
-                propValue -> apiValueToJsValue(f.value)
-            ))
-          .toVector)
+      propId -> value.tycon.map(_.toJson).getOrElse(JsNull),
+      propFields -> JsArray(value.fields.toSeq.zipWithIndex.map {
+        case ((oflabel, fvalue), ix) =>
+          JsObject(
+            propLabel -> JsString(oflabel getOrElse (ix: Int).toString),
```

Comment:
The idea of this codec was to have a fully self-contained representation of a DAML value, for consumers that don't have any access to the DAML types. Such consumers will probably not be happy with the numeric labels, so we could also consider blowing up instead.

However:
- there is no formal spec for this format
- in the context of navigator, the field names should always be available
- even if they weren't, it will only produce slightly awkward results in the front end, and not really break anything

So the code is fine as is.

Reply:
> in the context of navigator, the field names should always be available

Another important consideration was that numbers will never be field names, so these are safe unique fallbacks.